core/store/src/flat_state.rs

```diff
@@ -991,6 +1011,9 @@ impl FlatStorageState {
             };
         }
 
+        if let Some(value_ref) = guard.get_cached_ref(key) {
+            return Ok(value_ref);
+        }
         Ok(store_helper::get_ref(&guard.store, key)?)
```

Comment:
Should you also push the just read value back to the cache? Currently you only update values in the cache when writing new values there. I suppose either can be fine but cache hit rate will heavily depend on the usage pattern.

Reply:
Right!